### PR TITLE
feat: add controller associations support

### LINKED ISSUE #25: Add support to manage associations
We should expose the functionality to allow managing associations between nodes.

### PR DESCRIPTION
**Pull request by Michsior14**
Closes #25

### CODE REVIEW

File: src/lib/controller/message_handler.ts [hunk 1]
```diff
@@ -57,6 +57,48 @@ export class ControllerMessageHandler {
...
+          .forEach((value, key) => (associations[key] = value));
+        return { associations };
+      }
+      case ControllerCommand.isAssociationAllowed: {
+        const success = await driver.controller.isAssociationAllowed(
+          message.nodeId,
+          message.group,
+          message.association
+        );
+        return { success };
```
Comment (balloob) [inline src/lib/controller/message_handler.ts:80]: Should this be allowed?
```suggestion
        const allowed = await driver.controller.isAssociationAllowed(
          message.nodeId,
          message.group,
          message.association
        );
        return { allowed };
```

**Review by balloob — APPROVED** [review]
One comment. Rest looks great 🎉

File: src/lib/controller/outgoing_message.ts [hunk 2]
```diff
@@ -1,14 +1,25 @@
...
   [ControllerCommand.stopInclusion]: { success: boolean };
   [ControllerCommand.beginExclusion]: { success: boolean };
   [ControllerCommand.stopExclusion]: { success: boolean };
-  [ControllerCommand.removeFailedNode]: {};
+  [ControllerCommand.removeFailedNode]: Record<string, never>;
   [ControllerCommand.replaceFailedNode]: { success: boolean };
   [ControllerCommand.healNode]: { success: boolean };
   [ControllerCommand.beginHealingNetwork]: { success: boolean };
   [ControllerCommand.stopHealingNetwork]: { success: boolean };
   [ControllerCommand.isFailedNode]: { success: boolean };
```
Comment (Michsior14) [inline src/lib/controller/outgoing_message.ts:14]: @balloob Shouldn't we change the response here as well to i.e. `{ failed: boolean }`?

Comment (balloob) [reply src/lib/controller/outgoing_message.ts:14]: Yes we should.

Comment (balloob) [reply src/lib/controller/outgoing_message.ts:14]: Too many copy pasting 😅

**Comment by balloob** [comment]
Are you on Discord? Could you ping me, I'm @balloob there too.

**Pull request merged**